go/host/rpc/enclaverpc/enclave_client.go

```diff
@@ -352,25 +360,80 @@ func (c *Client) DebugTraceTransaction(hash gethcommon.Hash, config *tracers.Tra
 		return nil, err
 	}
 
-	resp, err := c.protoClient.DebugTraceTransaction(timeoutCtx, &generated.DebugTraceTransactionRequest{
+	response, err := c.protoClient.DebugTraceTransaction(timeoutCtx, &generated.DebugTraceTransactionRequest{
 		TxHash: hash.Bytes(),
 		Config: confBytes,
 	})
-	if err != nil {
-		return nil, err
+	if sysErr := c.handleErr(response, err); sysErr != nil {
+		return nil, sysErr
 	}
-	return json.RawMessage(resp.Msg), nil
+	return json.RawMessage(response.Msg), nil
 }
 
-func (c *Client) DebugEventLogRelevancy(hash gethcommon.Hash) (json.RawMessage, error) {
+func (c *Client) DebugEventLogRelevancy(hash gethcommon.Hash) (json.RawMessage, common.SystemError) {
 	timeoutCtx, cancel := context.WithTimeout(context.Background(), c.config.EnclaveRPCTimeout)
 	defer cancel()
 
-	resp, err := c.protoClient.DebugEventLogRelevancy(timeoutCtx, &generated.DebugEventLogRelevancyRequest{
+	response, err := c.protoClient.DebugEventLogRelevancy(timeoutCtx, &generated.DebugEventLogRelevancyRequest{
 		TxHash: hash.Bytes(),
 	})
-	if err != nil {
-		return nil, err
+	if sysErr := c.handleErr(response, err); sysErr != nil {
+		return nil, sysErr
+	}
+	return json.RawMessage(response.Msg), nil
+}
+
+// handleErr ensures to check for errors in both the RPC connection and in the RPC response object
+// creates different errors depending on the error found
+func (c *Client) handleErr(rpcResp interface{}, connError error) error {
+	if connError != nil {
+		c.logger.Error("enclave client RPC Connection err - ", log.ErrKey, connError)
+		return syserr.NewRPCError(connError)
+	}
+
+	if rpcResp != nil {
```

Comment:
I'm blind

Reply:
lol, yeah, it's not pretty. I either centralized the type casting for a specific type with this or make the code more verbose in all the methods.